pkg/parser/seven/parser.go

```diff
@@ -123,11 +125,22 @@ func (p *Parser) Decode(payload *pb.Payload, decoded *pb.Flow) error {
 		}
 	}
 
+	l4, sourcePort, destinationPort := decodeLayer4(r.TransportProtocol, sourceEndpoint, destinationEndpoint)
```

Comment:
nit: Is there a reason you chose `int` for the port number?

To my knowledge, there is no case where a port would not fit into 16 bits and Cilium always sends a uint16 anyways. I think it would save you a few casts if the ports returned here were `uint16` and `ServiceGetter.GetServiceByAddr()` would take a `uint16` as well.

Reply:
The reason I picked `int` is because Go's [`net`](https://golang.org/pkg/net/) package uses `int` for port. However, I can change it to `uint16`. Note though that I have seen `uint32` being used for the port in the codebase as well. On a 64-bit platform, `int` defaults to `int64` and thus `uint32` can fit into. However, if I change from `int` to `uint16` as you suggest, we can in theory run into problems with `uint32` ports.

Reply:
Ah! The `uint32` is due to  protobuf not having any 16bit integer types - I ideally we would have used `uint16` there as well, but we opted for uint32 instead: https://developers.google.com/protocol-buffers/docs/proto#scalar

 I leave it up to you if you want to change it. The Cilium service implementation itself is uses uint16 quite extensively, so using a `uint16` is basically guaranteed to be sufficient for the foreseeable future.

Reply:
I wasn't aware of this protobuf limitation and it looks like `uint16` is used consistently for port outside of code interacting with protobuf. I'll make the change to `uint16` then.

Reply:
Awesome, thanks!!